compute_sdk/tests/unit/test_executor.py

```diff
@@ -263,12 +263,43 @@ def test_map_raises(gc_executor):
         gce.map(noop)
 
 
+def test_reload_tasks_requires_task_group_id(gc_executor):
+    _gcc, gce = gc_executor
+
+    with pytest.raises(Exception) as e:
+        gce.reload_tasks()
+
+    assert "must specify a task_group_id in order to reload tasks" in str(e.value)
+
+
+@mock.patch(
+    "globus_compute_sdk.sdk.executor.Executor.task_group_id",
+    new_callable=mock.PropertyMock,
+)
+def test_reload_tasks_sets_passed_task_group_id(mock_task_group_id, gc_executor):
+    _gcc, gce = gc_executor
+
+    def raise_if_setting(*args):
+        if args:
+            raise Exception("bailing")
+
+    mock_task_group_id.side_effect = raise_if_setting
+
+    tg_id = str(uuid.uuid4())
+    with pytest.raises(Exception) as e:
+        gce.reload_tasks(tg_id)
+
+    assert "bailing" in str(e.value)
+    assert mock_task_group_id.called_with(tg_id)
+
```

Comment:
This took me a minute to grok why we want it to raise&nbsp;&mdash;&nbsp;to bail early so we don't have to mock out other items in the test.  Clever, but subtle.  You might consider a comment about this tactic and perhaps testing the setting more directly by mocking the following call instead.  Perhaps:
```python
def test_reload_tasks_sets_task_group_id(gc_executor):
    _gcc, gce = gc_executor

    gce.funcx_client.web_client.get_taskgroup_tasks = mock.Mock(side_effect=RuntimeError)

    assert gce.task_group_id is None, "Verify test setup"
    tg_id = str(uuid.uuid4())
    with pytest.raises(RuntimeError) as e:
        # just to bail early; less mocking needed
        gce.reload_tasks(tg_id)

    assert gce.task_group_id == tg_id
```